### PR TITLE
Add docs on how to run UI tests on android

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This adds the following docs to the docs folder:
### How to run UI tests for Android

There's only one dependency: having an actual already built and signed APK. The tests won't run on an unsigned apk.
It doesn't matter if the APK** is from a debug, release or adhoc build, it just has to be signed.

#### How to sign
For release builds, Bitrise has a step to sign the apk, but I don't think we should run UI tests on release builds. (Except for the Google-related tests, if we ever do them)
Don't run UI tests on production without approval from the team lead.
For adhoc and debug builds, any keystore would do and we can safely use the default android debug keystore for UI tests, which can be found on a secret folder inside your home folder if you have the SDK installed (either by having Android Studio, Xamarin, or just the Android SDK installed in your machine)

For adhoc and release builds, you can grab them from the latest builds on Bitrise.
For debug builds, you'll need to archive a debug build.

Debug keystore information (MacOS):
- Path: `~/.android/debug.keystore`
- Alias: `androiddebugkey`
- Store password: `android`
- Key password: `android`

On Visual Studio:
1. Select the Toggl.Giskard > Options > Android Build > Uncheck `Use Shared Mono Runtime`; 
2. Select from Visual Studio top menu: Build > Archive for publishing (this takes a while);
3. Now you have a new apk at `Toggl/bin/Debug/com.toggl.giskard.debug.apk`
4. You will have a new screen open, but you can get there by going to Build > ViewArchives;
5. Select the apk you just archived > Click in the bottom of the page "Sign and Distribute" > Ad Hoc > androiddebugkey (create it if you don't have it) > Next > Publish > Select a folder
5.1 If you don't have the androiddebugkey ready, click on "Import an Existing Key" and fill in the inputs with the values above. You can create a new one if you want, it doesn't matter much.
5.1.1 When creating a new signing key, you can choose the name, store password and key password as you wish. 
6. You have the signed APK
6.1 You'll want the apk without the -Signed suffix, if you try running the tests with it, you'll get the following error:
```
SetUp : System.Exception : Mono Shared Runtime is not supported. This can be resolved by changing the project configuration or using a Release build.
```

#### How to run the tests
First go to `Toggl.Giskard.Tests.UI/Configuration.cs` and update the path in `ApkFile` to point to your signed apk.
Ex: 
- if you put the apkName.apk file in your Desktop (MacOS) -> .ApkFile(`"/Users/yourusername/Desktop/apkName.apk"`)
- if you put the apk on (Repo root folder)/bin/Release/apkName.apk -> .ApkFile(`"../../bin/Release/apkName.apk"`)

##### To run on Visual Studio:
- Have the build target to Giskard | Debug | Any CPU
- On the right sidebar > Unit Tests > Giskard > Toggl.Giskard.Tests.UI
- On Test Apps select the device where you want to run the tests on, it can be an actual device or a running emulator.
- <a name="makingUITestsAppear"></a>If you don't see the tests there yet, double click Giskard > Toggl.Giskard.Tests.UI; fix whatever bugs might be there (the build might be broken) > click again > You'll eventually see all the tests from Shared.Toggl.Tests.UI;
- On Toggl.Tests.UI you can run all the tests or only the ones you want.
- That's it. 

##### To run on Rider:
- First thing you should know, switching back and forth from VS and Rider might break your build. Building the project again from the command line usually fixes it; if it doesn't, clean then build again. :see_no_evil:
- First run [this step from the instructions on how to run the tests on VS](#makingUITestsAppear)
- In rider you just navigate in the bottom bar "Unit Tests" are and add the tests you want to run or go directly to the test classes and run the tests from the left panel icon; You can also run all the tests in the file by clicking on it and selecting "Run Unit Tests";

Remember: you'll be modifying the test files, you don't need a new apk for each test run; you can keep using the same apk while updating the tests.
**: Please note that the apk architecture must match the device's architecture you are running (arm apks will only run on arm devices/emulators, for example).

## :squid: Permissions
🦑 whenever

### CODE REVIEW

File: docs/ui-tests.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+### How to run UI tests for Android
+
+There's only one dependency: having an actual already built and signed APK.
+It doesn't matter if the APK is from a debug, release or adhoc build, it just have to be signed.
+
+#### How to sign
+For release builds, our Bitrise process uses a step to sign it, but I don't think we should run UI tests on release builds. (Except by the Google-related tests, if we ever do them)
+For adhoc and debug builds, any keystore would do and we can safely use the default android debug keystore it for UI tests, which can be found on a secret folder inside your home folder if you have the SDK installed (either by having Android Studio, Xamarin, or just the Android SDK installed in your machine)
```
Comment (dshr) [inline docs/ui-tests.md]: > keystore it

an extra "it"?

File: docs/ui-tests.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+4. You will have a new screen open, but you can get there by going to Build > ViewArchives;
+5. Select the apk you just archived > Click in the bottom of the page "Sign and Distribute" > Ad Hoc > androiddebugkey (create it if you don't have it) > Next > Publish > Select a folder
+5.1 If you don't have the androiddebugkey ready, click on "Import an Existing Key" and fill in the inputs with the values above. You can create a new one if you want, it doesn't matter much.
+6. You have the signed debug APK (be careful, you want the apk without the -Signed suffix, it doesn't make sense... but the -Signed apk seems to have the mono runtime)! (or adhoc, the same process applies for it)
+
+#### How to run the tests
+First go to Toggl.Giskard.Tests.UI/Configuration.cs and update the path in `ApkFile` to point to your signed apk.
+
+##### To run on Visual Studio:
+Have the build target to Giskard ; Debug | Any CPU
```
Comment (dshr) [inline docs/ui-tests.md]: `;` -> `|` ?

File: docs/ui-tests.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+2. Select from Visual Studio top menu: Build > Archive for publishing (this takes a while);
+3. Now you have a new apk at `Toggl/bin/Debug/com.toggl.giskard.debug.apk`
+4. You will have a new screen open, but you can get there by going to Build > ViewArchives;
+5. Select the apk you just archived > Click in the bottom of the page "Sign and Distribute" > Ad Hoc > androiddebugkey (create it if you don't have it) > Next > Publish > Select a folder
+5.1 If you don't have the androiddebugkey ready, click on "Import an Existing Key" and fill in the inputs with the values above. You can create a new one if you want, it doesn't matter much.
+6. You have the signed debug APK (be careful, you want the apk without the -Signed suffix, it doesn't make sense... but the -Signed apk seems to have the mono runtime)! (or adhoc, the same process applies for it)
+
+#### How to run the tests
+First go to Toggl.Giskard.Tests.UI/Configuration.cs and update the path in `ApkFile` to point to your signed apk.
+
```
Comment (ApocalypticFish) [inline docs/ui-tests.md]: Can you include an example of the path .ApkFile("../../bin/Debug/com.toggl.giskard.debug-Signed.apk")

**Comment by daividssilverio** [comment]
I'll have to update the docs a bit... I run into some issues today when trying to run the tests on iOS to see if I had broken anything... Running on Rider isn't as simply as I thought... And the apk signing has to be clarified.

**Comment by daividssilverio** [comment]
@simonrozsival I researched a bit but I couldn't find why the apk has to be signed. I'll research a bit more later and try to add it to the docs. I'm more confused now because it seems that the tests are signing the apk when it's running... based on the console output...

**Comment by simonrozsival** [comment]
If it's necessary because it doesn't work without this step, but we don't know how it works internally, it would be OK to put it in the docs like that. We don't have to waste too much time on it if it's not documented.

File: docs/ui-tests.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+#### How to run the tests
+First go to Toggl.Giskard.Tests.UI/Configuration.cs and update the path in `ApkFile` to point to your signed apk.
+Ex: 
+- if you put the apkName.apk file in your Desktop (MacOS) -> `.ApkFile("/Users/yourusername/Desktop/apkName.apk")`
+- if you put the apk on (Repo root folder)/bin/Release/apkName.apk -> `.ApkFile("../../bin/Release/apkName.apk")`
+
+##### To run on Visual Studio:
+Have the build target to Giskard | Debug | Any CPU
+On the right sidebar > Unit Tests > Giskard > Toggl.Giskard.Tests.UI
+On Test Apps select the device where you want to run the tests (it can be an actual device or an emulator, but it has to be already running)
```
Comment (dshr) [inline docs/ui-tests.md]: Just tried this, and it seems to want an ARM-based EMU. Is that necessary?

Comment (daividssilverio) [reply docs/ui-tests.md]: No, but the apk you build must match the emulator/device architecture.

Comment (dshr) [reply docs/ui-tests.md]: I didn't see an architecture option when building the APK though. Let me check, I might've missed it.

Comment (dshr) [reply docs/ui-tests.md]: Nope, nothing there..

**Comment by heytherewill** [comment]
Why not have two docs: One for each platform? If you agree with that, can you rename the doc added by this PR to make it reference droid explicitly?

File: docs/running-android-ui-tests.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+### How to run UI tests for Android
+
+There's only one dependency: having an actual already built and signed APK.
+It doesn't matter if the APK** is from a debug, release or adhoc build, it just have to be signed.
+
+Why? It doesn't work otherwise. Further explanations why is like might be added at some point.
+
+#### How to sign
+For release builds, our Bitrise process uses a step to sign it, but I don't think we should run UI tests on release builds. (Except by the Google-related tests, if we ever do them)
```
[the remark this answers is not in the document]
Comment (amulware) [reply docs/running-android-ui-tests.md]: Very much agreed here. I don't want any random testing in production without approval.

[the remark this answers is not in the document]
Comment (daividssilverio) [reply docs/running-android-ui-tests.md]: And @kornelijepetak found a way to run Google with debug, so no need o/

File: docs/running-android-ui-tests.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+For adhoc and debug builds, any keystore would do and we can safely use the default android debug keystore for UI tests, which can be found on a secret folder inside your home folder if you have the SDK installed (either by having Android Studio, Xamarin, or just the Android SDK installed in your machine)
+
+For adhoc and release builds, you can grab them from the latest builds on Bitrise.
+For debug builds, you'll need to archive a debug build.
+
+Save this:
+- Path: `~/.android/debug.keystore`
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply docs/running-android-ui-tests.md]: Do you have it? Never developed android using Windows.

File: docs/ui-tests.md (hunk 4 above)
Comment (daividssilverio) [reply docs/ui-tests.md]: Are you building using VS, Rider? You can select specific Abis or using all in one in the project options on VS.

File: docs/running-android-ui-tests.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+- Alias: `androiddebugkey`
+- Store password: `android`
+- Key password: `android`
+
+On Visual Studio:
+1. Select the Toggl.Giskard > Options > Android Build > Uncheck `Use Shared Mono Runtime`;
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply docs/running-android-ui-tests.md]: It was for me, Idk why.

**Comment by ApocalypticFish** [comment]
It's looking pretty good and seems to work for me. Might be worth getting someone who doesn't have the tests running to try and give this a try just to make sure it all goes smoothly.

**Comment by heytherewill** [comment]
y u no merge dis

**Pull request merged**